### PR TITLE
For #12378: Build with compileSdkVersion for Android 13 / API 33.

### LINKED ISSUE #12378: Build with SDK for Android 13 / API 33
According to the [docs](https://developer.android.com/about/versions/13) has reached Beta 3, that "marks the [Platform Stability](https://developer.android.com/about/versions/13/overview#timeline) milestone, which means that Android 13 has reached final internal and external APIs and final app-facing behaviors".

We can try using 33 as `compileSdkVersion` to also be prepared for the upgrade of `targetSdkVersion`

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-20790)

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [x] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

Fixes #12378

### CODE REVIEW

File: components/feature/autofill/src/main/java/mozilla/components/feature/autofill/response/fill/AuthFillResponseBuilder.kt [hunk 1]
```diff
@@ -101,9 +102,22 @@ internal fun FillResponse.Builder.setAuthentication(
     inlinePresentation: InlinePresentation? = null,
     presentation: RemoteViews
 ): FillResponse.Builder {
-    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && inlinePresentation != null) {
+    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        val presentations: Presentations.Builder = Presentations.Builder()
+        presentations.apply {
+            inlinePresentation?.let {
+                setInlinePresentation(it)
+                setInlineTooltipPresentation(inlinePresentation)
```
Comment (mcarare) [inline components/feature/autofill/src/main/java/mozilla/components/feature/autofill/response/fill/AuthFillResponseBuilder.kt:110]: Are these supposed to be the same? Also, do both need to be set?

**Comment by mcarare** [comment]
bors try

**Comment by gabrielluong** [comment]
Can you spin out the ktlint change as a separate PR that we can land later? Since there's already numerous code changes, I don't think we should include ktlint changes in this PR.

**Comment by mcarare** [comment]
> Can you spin out the ktlint change as a separate PR that we can land later? Since there's already numerous code changes, I don't think we should include ktlint changes in this PR.

I could regenerate the baseline to include the new issues (or update references), but that would kind of defeat the purpose of a baseline (if we keep changing it, instead of fixing the issues).

**Comment by gabrielluong** [comment]
Hey @mcarare This project moved to a new repository. It is now developed and maintained at: https://github.com/mozilla-mobile/firefox-android

**Pull request closed without merging**